df_udf/src/main/scala/org/apache/spark/sql/nvidia/LogicalPlanRules.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.nvidia
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.expressions.{Expression, ScalaUDF}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.rules.Rule
+
+case class LogicalPlanRules() extends Rule[LogicalPlan] with Logging {
+  val replacePartialFunc: PartialFunction[Expression, Expression] = {
+    case f: ScalaUDF if DFUDF.getDFUDF(f.function).isDefined =>
+      DFUDF.getDFUDF(f.function).map {
+        dfudf => DFUDFShims.columnToExpr(
```

Comment:
Is this ever intended to run without the RAPIDS Accelerator (e.g.: CPU-only job, but using this extension to implement a UDF via DataFrame column APIs)?  If so, I'm not sure how the shims are supposed to work, since the RAPIDS Accelerator plugin sets up the necessary classloader for shims.

Reply:
Yes, it is intended to run on the CPU without the plugin. Right now there will be a separate jar for each version of Spark we want to support. I have verified that it works without the RAPIDS accelerator plugin, but I will do it again. So the issue is not really shims, but packaging it all into an uber-jar. That is what I intend to work on next in a follow on PR.

Reply:
OK that makes sense.  Running with the per-Spark jar doesn't use the parallel-worlds setup, so it doesn't need a custom classloader like an uber jar would require.